### DISCUSSION TITLE
The page keeps redirecting/reloading back to redirect uri then to requested page

### DISCUSSION THREAD

**Discussion by singhshubhjeet** [OpenIDC/mod_auth_openidc#910] [Q&A]
I have a loadbalancer in front of two servers where mod_auth_openidc-2.4.11.2-1.el7 is installed with below config file

LoadModule auth_openidc_module modules/mod_auth_openidc.so
OIDCProviderMetadataURL XXX/oauth2/.well-known/openid-configuration
OIDCClientID XXX
OIDCClientSecret XXX
OIDCScope "openid profile"

 OIDCRedirectURI /redirect_uri/
OIDCCryptoPassphrase XXX
OIDCResponseType "id_token"
<Location />
  AuthType openid-connect
 Require valid-user

</Location>
OIDCCacheEncrypt On
OIDCCacheShmEntrySizeMax 65536

When I try to access the url of individual server it works fine, but when I try to access it using loadbalancer url it keeps on looping on the same page. It works only for 1 out of 10 tries 

**Could you please help us understand why it works for the individual server url but not working with loadbalancer in place and how can we resolve this issue ?**

**Comment by zandbelt** [comment]
why use `OIDCResponseType "id_token"`? please remove it and try with the default flow; also, since you're using 2 different servers, you need to apply sticky session or share sessions using one of the mechanisms described in https://github.com/zmartzone/mod_auth_openidc/wiki/Caching

**Reply by singhshubhjeet** [reply]
even after removing "OIDCResponseType" and using below shared memory cache I am facing the same issue. Below is the conf added related to shm

 OIDCCacheType shm
 OIDCCacheShmMax 1000
OIDCCacheShmEntrySizeMax 65536

**Reply by zandbelt** [reply]
shm is not sharing across different Apache servers

**Reply by Amaury4sg** [reply]
Hello,
apologies we should have configured the OIDCSessionType client-cookie:persistent to get a stateless experience
It's working right now.
Sorry to have disturbed you.
shubhjeet you could set the question as closed
